### ISSUE TITLE
Add signature to all messages

### ISSUE THREAD

**Issue by enoch85** [owncloud/mail#654]
It would be nice to have the ability to add a signature to every mail, instead of having to write the same thing every time, for example:

```
Best Regards 
Daniel Hansson
www.en0ch.se
```

Like I suggested earlier to @jancborchardt it could be one of the things that would be integrated in a settings page.

It would also be cool to be able to have different signatures for different mailboxes.

**Comment by DeepDiver1975** [comment]
My pov:
One mailbox csn have multiple aliases and each alias has it's own signature.

**Comment by enoch85** [comment]
@DeepDiver1975 Yes, that´s like what I meant. Just expressed it the wrong way. :)

**Comment by enoch85** [comment]
@jancborchardt I think this should go in under 0.2

**Comment by jancborchardt** [comment]
Good call – basic feature.

**Comment by enoch85** [comment]
Would this be hard to implement, for me with no skills in coding?

**Comment by DeepDiver1975** [comment]
Because we want to link the signatures with the aliases which are available in one account - we basically need to:
- [ ] add a new table mail_aliases (columns: int id, int account_id, varchar alias_mail, varchar alias_display_name, text signature)
- [ ] add an alias entity
- [ ] add an alias entity mapper class
- [ ] add an alias controller
- [ ] add some dialog for settings
- [ ] append the signature to the new message composer based on the selected alias

**Comment by enoch85** [comment]
To much for me. :/

**Comment by jancborchardt** [comment]
Moving to 0.3 as the other stuff is more important for 0.3 – unless someone just wants to go ahead and implement it of course. ;)

**Comment by enoch85** [comment]
What a shame. This is the only thing left for me, using the mail app 100%. Would appriciate it very much if someone made it happen.

**Comment by jancborchardt** [comment]
Yeah – we don’t keep anyone from doing it of course. :) But stuff like forwarding and adding attachments from computer / to replies is definitely more important and we need to focus on that first.

**Comment by jancborchardt** [comment]
(Also, we have a release cycle of only 2 months so it won’t be long. ;)

**Comment by ronoliverclarin** [comment]
Craving for this feature.

**Comment by enoch85** [comment]
@jancborchardt See this feature as a bit of commercial for ownCloud mail. Because it could say something like:
```
-- Sent from ownCloud mail. [link]
```

**Comment by jancborchardt** [comment]
Yeah, but as you all know these signatures are damn annoying. So I have no rush in getting an annoying email signature into circulation which screams our product name everywhere even though people receiving emails don’t really care. ;)

**Comment by enoch85** [comment]
@jancborchardt Depends on who you ask ;) But ok, I get your point. 

Btw, see you at the conference!

**Comment by siemendev** [comment]
When will the implementation of this feature happen?
I really want to migrate from Afterlogic Webmail to this App because it´s better integrated into owncloud, but without the ability to add Signatures your App is worthless for me.
So +1 for these Issue! :+1:

**Comment by jancborchardt** [comment]
@siemendev all depends on someone doing the work. :) So seeing how you have some dev proficiency, do you want to dig in here? I guess @ChristophWurst @irgendwie @Gomez can show you some pointers.

**Comment by siemendev** [comment]
So if some one introduces me where to find what, I would really like to try out implementing it!
Just contact me via post@patrick-siemen.de with the necessary information :-)

**Comment by DeepDiver1975** [comment]
> Just contact me via post@patrick-siemen.de with the necessary information :-)

Please stop by on irc #owncloud-mail on freenode - this is a better place to communicate

**Comment by Gomez** [comment]
@siemendev Great! This feature is really needed. You can setup the dev environment and start looking into the code. If you have any questions come to irc or ask here / in a Pull Request.
